packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_controller.dart

```diff
@@ -89,32 +211,193 @@ class DeepLinksController {
         .toList();
   }
 
-  final selectedProject = ValueNotifier<FlutterProject?>(null);
+  final selectedProject = ValueNotifier<FlutterProject?>( null);
   final selectedLink = ValueNotifier<LinkData?>(null);
-  final linkDatasNotifier = ValueNotifier<List<LinkData>?>(null);
-  final showSpitScreenNotifier = ValueNotifier<bool>(false);
-
-  final _searchContentNotifier = ValueNotifier<String>('');
-
-  void _updateLinks() {
-    final searchContent = _searchContentNotifier.value;
-    final List<LinkData> linkDatas = searchContent.isNotEmpty
-        ? _allLinkDatas
-            .where(
-              (linkData) => linkData.matchesSearchToken(
-                RegExp(
-                  searchContent,
-                  caseSensitive: false,
-                ),
-              ),
-            )
-            .toList()
-        : _allLinkDatas;
-    linkDatasNotifier.value = linkDatas;
+
+  final allLinkDatasNotifier = ValueNotifier<List<LinkData>?>(null);
+  final displayLinkDatasNotifier = ValueNotifier<List<LinkData>?>(null);
+  final generatedAssetLinksForSelectedLink = ValueNotifier<String?>(null);
+
+  final displayOptionsNotifier =
+      ValueNotifier<DisplayOptions>(DisplayOptions());
+
+  Future<void> _generateAssetLinks() async {
+    final applicationId =
+        _androidAppLinks[selectedVariantIndex.value]?.applicationId ?? '';
+
+    final response = await http.post(
+      Uri.parse(
+        'https://deeplinkassistant-pa.googleapis.com/android/generation/v1/assetlinks:generate?key=$_apiKey',
+      ),
+      headers: {'Content-Type': 'application/json'},
+      body: jsonEncode(
+        {
+          'package_name': applicationId,
+          'domains': [selectedLink.value!.domain],
```

Comment:
put these keys into named constants and use everywhere we are using the raw strings